fix/simplify/speed-up sandbox

tophf opened the pull request:
* added the missing methods to `unforgeables` and `readonlys`

* Fixed an exception when a userscript wanted a particular injection mode (in its meta or VM's global mode) but the mode was non-injectable.

* `unforgeables` is defined now in the wrapper's init to let JS natively handle subsequent attempts to override it
* `readonlys` is correctly depopulated on redefines
* `globalKeys` is used again to prevent userscripts from seeing page vars directly

* simplified stepping though code in debugger by removing the need for webpack import wrappers like `n.d(t, "d", ()=>s)` so stepping into a sandboxed function call like `console.log(document)` will require pressing F11 key only 21 times instead of 42.

* removed an unsafe call to regexp#test in isUnforgeableFrameIndex and fixed it so only int32 numbers are handled.
* inlined single-use imports in helpers.js

* Switched to TextDecoder so now GM_getResourceText will be ~15 times faster: 0.15s instead of 2.5s on 9MB text.
* Reverted to plain INIT_FUNC_NAME because Symbol.for function can be spoofed.
* fixes #916.
* more trivial fixes and speedups

tophf commented:
@gera2ld the build failure is caused by something on their server:

```
gpg: BAD signature from "Yarn Packaging <yarn@dan.cx>"
> GPG signature for this Yarn release is invalid! This is BAD and may mean the release has been tampered with. It is strongly recommended that you report this to the Yarn developers.
The command "curl -o- -L https://yarnpkg.com/install.sh | bash" failed and exited with 1 during .
```

gera2ld commented:
> simplified stepping though code in debugger by removing the need for webpack import wrappers like `n.d(t, "d", ()=>s)` so stepping into a sandboxed function call like `console.log(document)` will require pressing F11 key only 21 times instead of 42.

Just curious, where should I put the `debugger` to reproduce this?

tophf commented:
Set a breakpoint on `console.log(document)` and when it hits, step through by pressing F11.

tophf commented:
Hmm, I have an idea: now that `globalKeys` are back I can probably get rid of `deleted` and maybe even switch to `Map` as its lookup should be faster than both Object::has or Array::includes. Gonna try that.

tophf commented:
Done.

tophf commented:
Switched to an array+join instead of progressive concatenation in GM_getResourceText so now it's ~2.5 times faster: 1.1sec to decode 9MB text instead of 2.7sec.

tophf commented:
* Switched to TextDecoder so now GM_getResourceText will be ~15 times faster: 0.15 s instead of 2.5 s on 9MB text.
* Reverted to plain INIT_FUNC_NAME because Symbol.for function can be spoofed.
